src/session/Session.java

```diff
@@ -0,0 +1,53 @@
+package session;
+
+import encoder.StringEncoder;
+
+import java.security.MessageDigest;
+
+public class Session {
+    private Repository repo;
+    private Executor exec;
+    private BankClient user;
+
+    public Session() {
+        this.repo = new Repository();
+    }
+
+    public Executor getExecutor() {
+        return exec;
+    }
+
+    public Repository getRepository() {
+        return repo;
+    }
+
+    public List<UserBankAccount> getUserBankAccounts() {
+        return repo.getUserBankAccountsById(user.getId());
+    }
+
+    public void login(int id, String pass) {
+        BankClient user = repo.getBankClientById(id);
+        if(this.checkValidity(user, pass))
+            initializeAfterLogin(user);
+        else
+            throw new IllegalArgumentException("Invalid password.");
+    }
+
+    private boolean checkValidity(BankClient user, String pass) {
+        if (user == null) {
+            throw new IllegalArgumentException("Invalid login.");
+        } else {
+            pass = user.hashFunc(pass);
+            return user.getPassword.equals(pass);
+        }
+    }
```

Comment:
Może troche czepianie się ale nie powinno wypisywać że źle jest tylko login lub hasło bo to jest nieznaczna luka w bezpieczenstwie. Powinno być że " invalid login or password".